JavaScript/JavaScriptSDK.Tests/CheckinTests/Context/Session.tests.ts

```diff
@@ -376,5 +466,67 @@ class SessionContextTests extends TestClass {
     private restoreFakeCookie() {
         Microsoft.ApplicationInsights.Util["document"] = this.originalDocument;
     }
+
+    // Make a fake document.cookie property that correctly handles its unique getter/setter functionality.
+    // Document.cookie lets you read the entire set of cookies, but on set acts essentially as +=, with
+    // some logic to overwrite existing cookies with the same name.
+    // Document.cookie also silently removes expired cookies.
+    private generateEmulatedCookieStore() {
+        var cookieStore = {
+            _cookies: <string[]>[]
+        };
+        Object.defineProperty(cookieStore, "cookie", {
```

Comment:
This looks like well written code. Do you feel its appropriate to create an emulator for use in a single test? Emulators, such as this, are complex enough to prevent the tests from being [self-checking](http://xunitpatterns.com/Goals%20of%20Test%20Automation.html#Self-Checking Test) and often require separate testing of the emulation logic itself. Have you considered using plain stubs and spies instead?